README.md

```diff
@@ -0,0 +1,9 @@
+# Universal Packaging Tool: MacPorts backend
+This is a [MacPorts](https://www.macports.org) backend for [upt](https://pypi.python.org/pypi/upt).
+
+#### Steps to use this tool out:
+
+- Install upt
+- Clone this repository and run ```python3 setup.py install``` to install macports backend
+- Run ```$ upt package -f pypi -b macports package_name```
```

Comment:
@mojca I'm getting an error when executing this line: `UnboundLocalError: local variable 'upt_pkg' referenced before assignment`

Reply:
@rajdeepbharati: Thank you very much for testing.

Usually it helps to provide slightly more background information, so that it's easier to reproduce. I checked the code and tried to guess what could have happened and got some success after a while.

@Steap, this is "full trace" of what I get when using invalid package name:
```
$ upt package -f pypi -b macports invalid_package_name

The package invalid_package_name could not be found by backend pypi
Traceback (most recent call last):
  File "/path/to/lib/python3.7/site-packages/upt-0.6-py3.7.egg/upt/upt.py", line 269, in main
    upt_pkg = frontend.parse(args.package)
  File "/path/to/lib/python3.7/site-packages/upt_pypi/upt_pypi.py", line 370, in parse
    raise upt.InvalidPackageNameError(self.name, pkg_name)
upt.exceptions.InvalidPackageNameError: The package invalid_package_name could not be found by backend pypi

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/path/to/lib/python3.7/site-packages/upt-0.6-py3.7.egg/upt/upt.py", line 278, in main
    sys.exit(1)
SystemExit: 1

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/path/to/bin/upt", line 11, in <module>
    load_entry_point('upt==0.6', 'console_scripts', 'upt')()
  File "/path/to/lib/python3.7/site-packages/upt-0.6-py3.7.egg/upt/upt.py", line 280, in main
    upt_pkg._clean()
UnboundLocalError: local variable 'upt_pkg' referenced before assignment
```

I suspect the problem is in cleaning up the object while handling the error occurring during object constructor, so the object itself didn't even exist yet. If that's the same error that @rajdeepbharati experienced, it's definitely something that I would consider a bug in `upt` itself, something outside of the scope of this PR.

@rajdeepbharati: are you also unable to use the tool with valid package names? I tend to agree that the above command doesn't really make it clear that `package_name` should be replaced with the actual name of some package. It could have been an argument that needed to be specified literally. In a weird coincident though this seems to be a valid package name, so even when copy-pasting the command one should get some semi-(in)valid package: https://pypi.org/project/package_name/

Reply:
@mojca When I am using a valid package name, I'm getting this error (this one is slightly different from the one when using an invalid package name). I used a python virtual environment. Here is the stack trace:
```
$ upt package -f pypi -b macports package_name
Traceback (most recent call last):
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/urllib/request.py", line 1317, in do_open
    encode_chunked=req.has_header('Transfer-encoding'))
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/http/client.py", line 1229, in request
    self._send_request(method, url, body, headers, encode_chunked)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/http/client.py", line 1275, in _send_request
    self.endheaders(body, encode_chunked=encode_chunked)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/http/client.py", line 1224, in endheaders
    self._send_output(message_body, encode_chunked=encode_chunked)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/http/client.py", line 1016, in _send_output
    self.send(msg)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/http/client.py", line 956, in send
    self.connect()
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/http/client.py", line 1392, in connect
    server_hostname=server_hostname)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/ssl.py", line 412, in wrap_socket
    session=session
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/ssl.py", line 853, in _create
    self.do_handshake()
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/ssl.py", line 1117, in do_handshake
    self._sslobj.do_handshake()
ssl.SSLCertVerificationError: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1056)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/rajdeep/upt-macports/.venv/lib/python3.7/site-packages/upt-0.6-py3.7.egg/upt/upt.py", line 269, in main
    upt_pkg = frontend.parse(args.package)
  File "/Users/rajdeep/upt-macports/.venv/lib/python3.7/site-packages/upt_pypi/upt_pypi.py", line 380, in parse
    'licenses': self.guess_licenses(self.json['releases'][version]),
  File "/Users/rajdeep/upt-macports/.venv/lib/python3.7/site-packages/upt_pypi/upt_pypi.py", line 328, in guess_licenses
    return self.guess_licenses_from_sdist(archive_url)
  File "/Users/rajdeep/upt-macports/.venv/lib/python3.7/site-packages/upt_pypi/upt_pypi.py", line 346, in guess_licenses_from_sdist
    urlretrieve(archive_url, archive.name)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/urllib/request.py", line 247, in urlretrieve
    with contextlib.closing(urlopen(url, data)) as fp:
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/urllib/request.py", line 222, in urlopen
    return opener.open(url, data, timeout)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/urllib/request.py", line 525, in open
    response = self._open(req, data)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/urllib/request.py", line 543, in _open
    '_open', req)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/urllib/request.py", line 503, in _call_chain
    result = func(*args)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/urllib/request.py", line 1360, in https_open
    context=self._context, check_hostname=self._check_hostname)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/urllib/request.py", line 1319, in do_open
    raise URLError(err)
urllib.error.URLError: <urlopen error [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1056)>

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/rajdeep/upt-macports/.venv/bin/upt", line 11, in <module>
    load_entry_point('upt==0.6', 'console_scripts', 'upt')()
  File "/Users/rajdeep/upt-macports/.venv/lib/python3.7/site-packages/upt-0.6-py3.7.egg/upt/upt.py", line 280, in main
    upt_pkg._clean()
UnboundLocalError: local variable 'upt_pkg' referenced before assignment
```

Is there some issue with my Python installation?

Reply:
Probably missing certificates? Try installing curl-ca-bundle or certsync.

Reply:
OK, I've fixed it upstream, even though a new version has not been released yet. Just be careful in the meantime :)

Reply:
Thanks @jmroot, it worked after installing curl-ca-bundle

Reply:
Thank you very much for the feedback.

Maybe we should better add a runtime dependency on `curl-ca-bundle` to one of these ports, not sure which one.